### PR TITLE
Sprint2 47 documentation

### LINKED ISSUE #47: Weaver Components Should have a documentation strategy
- Usage docs
- Development docs
- index page to choose between the two doc types
- github pages

### PR DESCRIPTION
**Pull request by rmathew1011**
Resolves #47

### CODE REVIEW

File: projects/wvr-elements/src/lib/wvr-nav-list/wvr-nav-li/wvr-nav-li.component.ts [hunk 1]
```diff
@@ -1,19 +1,24 @@
-import { Component, Input } from '@angular/core';
-import { wvrThemeHostBindings, wvrThemeInputs } from '../../shared/wvr-styles';
-
+import { Component, HostBinding, Input } from '@angular/core';
+/**
+ * The WvrNavLi Component contains the href property of the list.
```
Comment (jeremythuff) [inline projects/wvr-elements/src/lib/wvr-nav-list/wvr-nav-li/wvr-nav-li.component.ts]: The WvrNavLi Component is the list element to be used with the wvr-nav-list element.

File: projects/wvr-elements/src/lib/wvr-nav-list/wvr-nav-li/wvr-nav-li.component.ts [hunk 2]
```diff
@@ -1,19 +1,24 @@
...
-    ...wvrThemeInputs
-  ],
-  host: {
-    ...wvrThemeHostBindings
-  }
+  styleUrls: ['./wvr-nav-li.component.scss']
 })
 export class WvrNavLiComponent {
 
+  /** This property contains the href value of the list element. */
```
Comment (jeremythuff) [inline projects/wvr-elements/src/lib/wvr-nav-list/wvr-nav-li/wvr-nav-li.component.ts]: A resolvable URI to to which this li will link, when provided. If no link is provided, the list element will not display as a link.

File: projects/wvr-elements/src/lib/wvr-nav-list/wvr-nav-list.component.ts [hunk 3]
```diff
@@ -1,22 +1,19 @@
 import { Component, Input } from '@angular/core';
 import { Alignment } from '../shared/alignment.enum';
-import { wvrThemeHostBindings, wvrThemeInputs } from '../shared/wvr-styles';
-
+/**
+ * The WvrNavList Component contains the aligned and vertical properties that describe the positioning of the ul list.
```
Comment (jeremythuff) [inline projects/wvr-elements/src/lib/wvr-nav-list/wvr-nav-list.component.ts]: The WvrNavList Component presents a navigation list. Elements within this list must be wvr-nav-li elements and can be either links of action elements.

File: projects/wvr-elements/src/lib/wvr-nav-list/wvr-nav-list.component.ts [hunk 4]
```diff
@@ -1,22 +1,19 @@
...
-  }
+  styleUrls: ['./wvr-nav-list.component.scss']
 })
 export class WvrNavListComponent {
 
+  /** The aligned property describing the positioning of the list elements. */
   @Input() aligned = Alignment.LEFT;
 
-  @Input() vertical = false;
+  /** This boolean vertical property describing if the list can be displayed horizontally or vertically. */
```
Comment (jeremythuff) [inline projects/wvr-elements/src/lib/wvr-nav-list/wvr-nav-list.component.ts]: Toggles the display of the list horizontally or vertically.

File: projects/wvr-elements/src/lib/wvr-header/wvr-header.component.ts [hunk 5]
```diff
@@ -1,28 +1,41 @@
-import { Component, Input } from '@angular/core';
-import { DomSanitizer } from '@angular/platform-browser';
-import { wvrThemeHostBindings, wvrThemeInputs } from '../shared/wvr-styles';
+import { Component, HostBinding, Input } from '@angular/core';
+import { DomSanitizer, SafeUrl } from '@angular/platform-browser';
 
+/**
+ * @component This is the main header component. This contains elements like logo text, header title, logo image and the link to it.
```
[the remark this answers is not in the document]
Comment (rmathew1011) [reply projects/wvr-elements/src/lib/wvr-header/wvr-header.component.ts]: Updated the component description with the text provided above

File: projects/wvr-elements/src/lib/wvr-header/wvr-header.component.ts [hunk 6]
```diff
@@ -1,28 +1,41 @@
...
-    ...wvrThemeInputs
-  ],
-  host: {
-    ...wvrThemeHostBindings
-  }
+  styleUrls: ['./wvr-header.component.scss']
 })
 export class WvrHeaderComponent {
 
+  /** The text value to be displayed in the logo text. */
```
[the remark this answers is not in the document]
Comment (rmathew1011) [reply projects/wvr-elements/src/lib/wvr-header/wvr-header.component.ts]: Applied correction

File: projects/wvr-elements/src/lib/wvr-header/wvr-header.component.ts [hunk 7]
```diff
@@ -1,28 +1,41 @@
...
   @Input() logoText = 'Weaver Components';
 
+  /** The header title value to be displayed in the title section. */
```
[the remark this answers is not in the document]
Comment (rmathew1011) [reply projects/wvr-elements/src/lib/wvr-header/wvr-header.component.ts]: Applied correction

File: projects/wvr-elements/src/lib/wvr-header/wvr-header.component.ts [hunk 8]
```diff
@@ -1,28 +1,41 @@
...
+  /** The image to be displayed as the logo. */
```
[the remark this answers is not in the document]
Comment (rmathew1011) [reply projects/wvr-elements/src/lib/wvr-header/wvr-header.component.ts]: Applied correction

File: projects/wvr-elements/src/lib/wvr-header/wvr-header.component.ts [hunk 9]
```diff
@@ -1,28 +1,41 @@
...
 
+  /** The href value of the logo. */
```
[the remark this answers is not in the document]
Comment (rmathew1011) [reply projects/wvr-elements/src/lib/wvr-header/wvr-header.component.ts]: Applied correction

File: projects/wvr-elements/src/lib/wvr-header/wvr-header.component.ts [hunk 10]
```diff
@@ -1,28 +1,41 @@
...
   @Input() logoHref = '#test';
 
+  /** This field has hostbinding property bound to the  weaver --wvr-gray variable  - equivalent of css color of gray shade */
```
[the remark this answers is not in the document]
Comment (rmathew1011) [reply projects/wvr-elements/src/lib/wvr-header/wvr-header.component.ts]: Applied correction

File: projects/wvr-elements/src/lib/wvr-lib.module.ts [hunk 11]
```diff
@@ -24,6 +26,7 @@ const components = [
   WvrTextComponent
 ];
 
+/** This decorator helps organize the weaver component project and provides various features as defined in the properties below. */
```
[the remark this answers is not in the document]
Comment (rmathew1011) [reply projects/wvr-elements/src/lib/wvr-lib.module.ts]: Removed comment

File: projects/wvr-elements/src/lib/wvr-lib.module.ts [hunk 12]
```diff
@@ -16,6 +17,7 @@ const elements = [
   { component: WvrTextComponent, selector: 'wvr-text' }
 ];
 
+/** This property contains a list of components class. */
```
[the remark this answers is not in the document]
Comment (rmathew1011) [reply projects/wvr-elements/src/lib/wvr-lib.module.ts]: Applied correction

File: projects/wvr-elements/src/lib/wvr-lib.module.ts [hunk 13]
```diff
@@ -41,14 +44,15 @@ const components = [
   ],
   schemas: [CUSTOM_ELEMENTS_SCHEMA]
 })
+/** This is an NgModule genereated with the NgModule decorator. */
```
[the remark this answers is not in the document]
Comment (rmathew1011) [reply projects/wvr-elements/src/lib/wvr-lib.module.ts]: Applied correction

File: projects/wvr-elements/src/lib/wvr-nav-list/wvr-nav-li/wvr-nav-li.component.ts (hunk 1 above)
Comment (rmathew1011) [reply projects/wvr-elements/src/lib/wvr-nav-list/wvr-nav-li/wvr-nav-li.component.ts]: Applied correction for this comment and also for the remainder properties as well

File: projects/wvr-elements/src/lib/wvr-nav-list/wvr-nav-li/wvr-nav-li.component.ts (hunk 2 above)
Comment (rmathew1011) [reply projects/wvr-elements/src/lib/wvr-nav-list/wvr-nav-li/wvr-nav-li.component.ts]: Applied correction

File: projects/wvr-elements/src/lib/wvr-nav-list/wvr-nav-list.component.ts (hunk 3 above)
Comment (rmathew1011) [reply projects/wvr-elements/src/lib/wvr-nav-list/wvr-nav-list.component.ts]: Applied correction

File: projects/wvr-elements/src/lib/wvr-nav-list/wvr-nav-list.component.ts (hunk 4 above)
Comment (rmathew1011) [reply projects/wvr-elements/src/lib/wvr-nav-list/wvr-nav-list.component.ts]: Applied correction

File: projects/wvr-elements/src/lib/wvr-it-works/wvr-it-works.component.ts [hunk 14]
```diff
@@ -1,21 +1,21 @@
-import { Component, Input } from '@angular/core';
-import { wvrThemeHostBindings, wvrThemeInputs } from '../shared/wvr-styles';
-
+import { Component, HostBinding, Input } from '@angular/core';
+/**
+ * This is the it-works component. This contains a text property.
```
[the remark this answers is not in the document]
Comment (rmathew1011) [reply projects/wvr-elements/src/lib/wvr-it-works/wvr-it-works.component.ts]: Applied correction

File: projects/wvr-elements/src/lib/shared/alignment.enum.ts [hunk 15]
```diff
@@ -1,3 +1,3 @@
 export enum Alignment {
```
[the remark this answers is not in the document]
Comment (rmathew1011) [reply projects/wvr-elements/src/lib/shared/alignment.enum.ts]: Added comment

**Pull request merged**